### PR TITLE
Clean up stale metadata registrations.

### PR DESCRIPTION
**Pull request by pchote**

### CODE REVIEW

File: OpenRA.Game/ExternalMods.cs [hunk 1]
```diff
@@ -111,6 +111,34 @@ internal void Register(Manifest mod)
 			Directory.CreateDirectory(supportPath);
 
 			File.WriteAllLines(Path.Combine(supportPath, key + ".yaml"), yaml.ToLines(false).ToArray());
+
+			// Clean up stale mod registrations:
+			//  - LaunchPath no longer exists (uninstalled)
+			//  - LaunchPath and mod match the current mod, but version differs (newer version installed on top)
+			List<string> toRemove = null;
```
Comment (abcdefg30) [inline OpenRA.Game/ExternalMods.cs]: What is this for? I only see you adding things to it, but never actually doing something with those.

**Comment by pchote** [comment]
Fixed.  I also added an extra fix for the game server compatibility check, which I missed in my fixup of #12600.

**Review by abcdefg30 — APPROVED** [review]

**Review by phrohdoh — APPROVED** [review]
LGTM

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/e6c4bc5788908f9c19ba179ee4a29acd2bdb9237...b19f51350713d3ce0bbf0037eedf612859bfb4ad)

**Pull request merged**